src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,163 @@
+        return new SingleShotCommand(this) {
+            @Override
+            public void doCommand() {
+                double desired = hoodMotorController.getActualPosition() - 150;
+                adjustHoodPosition(desired);
+                while (!(Math.abs(desired - hoodMotorController.getActualPosition()) <= 5)) {
+
+                }
+                hoodMotorController.setDesiredPosition(0);
+                hoodMotorController.resetPosition();
+            }
+        }.withTimeout(EntechCommandBase.DEFAULT_TIMEOUT_SECONDS);
+    }
+
+    @Override
+    public void initialize() {
+
+        SparkMaxSettings shooterSettings = SparkMaxSettingsBuilder.defaults().withCurrentLimits(35).coastInNeutral()
+                .withDirections(false, true).noMotorOutputLimits().noMotorStartupRamping().useSpeedControl().build();
+        TalonSettings hoodSettings = TalonSettingsBuilder.defaults().withPrettySafeCurrentLimits().brakeInNeutral()
+                .withDirections(false, false).noMotorOutputLimits().noMotorStartupRamping().usePositionControl()
+                .withGains(PID_F, PID_P, PID_I, PID_D)
+                .withMotionProfile(HOOD_CRUISE_VELOCITY, HOOD_ACCELERATION, ALLOWABLE_ERROR).build();
+
+        // Basic outline for shooter
+        shooterSettings.configureSparkMax(shootMotor);
+
+        hoodMotorController = new TalonPositionController(hoodMotor, hoodSettings);
+        hoodMotorController.configure();
+        hoodMotor.configForwardLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen,
+                0);
+
+        hoodMotor.configReverseLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen,
+                0);
+
+        hoodMotor.overrideLimitSwitchesEnable(true);
+        HOOD_POSITION = hoodMotorController.getActualPosition();
+    }
+
+    public void adjustShooterSpeed(double desiredSpeed) {
+        logger.log("Intake Motor speed", desiredSpeed);
+        this.SHOOT_SPEED = desiredSpeed;
+        shootMotor.set(this.SHOOT_SPEED);
+    }
+
+    public void adjustHoodPosition(double desiredPosition) {
+        this.HOOD_POSITION = desiredPosition;
+        hoodMotorController.setDesiredPosition(desiredPosition);
+    }
+
+    public boolean isUpperLimitHit() {
+        return hoodMotor.getSensorCollection().isFwdLimitSwitchClosed();
+    }
+
+    public boolean isLowerLimitHit() {
+        return hoodMotor.getSensorCollection().isRevLimitSwitchClosed();
+    }
+
+    public void shoot() {
+        // There is still uncertainty here
+    }
+
+    public double getShooterSpeed() {
+        return shootMotor.getEncoder().getVelocity();
+    }
+
+    private static class LimitSwitchState {
+
+        public static int closed = 1;
+        public static int open = 0;
+    }
+
+    public double getHoodPosition() {
+        return hoodMotorController.getActualPosition();
+    }
+
+    public double getDesiredPositon() {
```

Comment:
if HOOD_POSITION is a constant, it should be a public static final. If it is not a constant, it should not be in all caps